src/Python/divik2spectre.py

```diff
@@ -0,0 +1,149 @@
+    :param location: 2D coordinates of the spectrum
+    :param region: assignment of spectrum to a Region-of-Interest
+    :return: text file entry
+    """
+    return '\n'.join([
+        ' '.join(map(str, [*location, CONST_Z_COORDINATE, region])),
+        ' '.join(map(str, spectrum))
+    ])
+
+
+class DivikDataSerializer(object):
+    """Serializes DiviK data into Spectre txt format"""
+    def __init__(self, data, xy, mz, region, region_names):
+        self.data = data
+        self.xy = xy
+        self.mz = mz.reshape((-1,))
+        self.region = region.reshape((-1))
+        self.region_names = region_names
+        self._serialized = None
+
+    def __str__(self):
+        if self._serialized is None:
+            header = '\n'.join([
+                'region names: ' + ', '.join(self.region_names),
+                ' '.join(map(str, self.mz))
+            ])
+            self._serialized = '\n'.join([
+                header,
+                *[_serialize_entry(spectrum, location, region)
+                  for spectrum, location, region
+                  in zip(tqdm(self.data, desc='Spectrum'), self.xy, self.region)]
+            ])
+        return self._serialized
+
+
+def _parse_hdf5_cell_array(object_reference, matfile) -> List[str]:
+    """Parses MATLAB cell array of character arrays from hdf5 file
+
+    :param object_reference: cell array in the file
+    :param matfile: mat-file parsed
+    :return: list of strings
+    """
+    data = []
+    for column in object_reference:
+        row_data = []
+        for row_number in range(len(column)):
+            row_data.append(''.join(map(chr, matfile[column[row_number]][:])))
+        data.append(row_data)
+    data = list(np.transpose(data)[0])
+    return data
+
+
+def read_hdf5(path: str) -> DivikDataSerializer:
+    """Parses hdf5 mat-file with DiviK data into serializer"""
+    matfile = h5py.File(path, 'r')
+    region_names = _parse_hdf5_cell_array(matfile['region_names'], matfile)
+    data = np.transpose(matfile['data'])
+    xy = np.transpose(matfile['xy']).astype(int)
+    assert data.shape[0] == xy.shape[0]
+    assert xy.shape[1] == 2
+    region = np.transpose(matfile['region']).astype(int)
+    assert region.size == data.shape[0]
+    if 'mz' not in matfile:
+        print('M/Z array not found, creating fake...')
+        mz = np.arange(data.shape[1])
+    else:
+        mz = np.transpose(matfile['mz'])
+        assert mz.size == data.shape[1]
+    return DivikDataSerializer(data, xy, mz, region, region_names)
+
+
+def read_matfile(path: str) -> DivikDataSerializer:
+    """Parses v7.0 >= mat-file with DiviK data into serializer"""
+    matfile = scio.loadmat(path)
+    region_names = [name_table[0] for name_table in matfile['region_names'][0]]
+    data = matfile['data']
+    xy = matfile['xy']
+    assert data.shape[0] == xy.shape[0]
+    assert xy.shape[1] == 2
+    region = matfile['region']
+    assert region.size == data.shape[0]
```

Comment:
The whole script looks solid. No tests though :P

Reply:
I know. This is due to several reasons:
1) There's only one functionality, that could be unit-tested properly: serializer. Rest of the script is I/O or file format related (I don't know the internals of MAT-file in a way I could mock it), so it would require hacky testing.
2) White-box testing was applied through `asserts` wherever I've seen such functionality that is untestable from the outside (through black-box testing).
3) I've created a kind of integration test - the validator application, which for me served *only* for validation of this script. I just thought it may be useful in the long run for the users also.

Reply:
SGTM.